### PR TITLE
Correct docker build

### PR DESCRIPTION
**Pull request by pletessier**
Hi,

Here is a correction solving the broken docker build. The solution is mainly based on the work of @beauby : https://gist.github.com/beauby/813791b8d511d7801c85f4cf73bc6307

@mdouze, it could solve issues #470 #491.

Pierre

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,37 +1,29 @@
...
+    apt-get install -y libopenblas-dev python-numpy python-dev swig python-pip curl
 
 RUN pip install matplotlib
 
 COPY . /opt/faiss
 
 WORKDIR /opt/faiss
 
-ENV BLASLDFLAGS /usr/lib/libopenblas.so.0
+ENV BLASLDFLAGS=/usr/lib/libopenblas.so.0 \
```
Comment (beauby) [inline Dockerfile]: This is a no-op.

File: Dockerfile [hunk 2]
```diff
@@ -1,37 +1,29 @@
...
+    PYTHONPATH=$PYTHONPATH:/opt/faiss \
```
Comment (beauby) [inline Dockerfile]: This shouldn't be needed since `make install` is called.

File: Dockerfile [hunk 3]
```diff
@@ -1,37 +1,29 @@
...
+    PYTHON=python
```
Comment (beauby) [inline Dockerfile]: This should not be needed.

File: Dockerfile [hunk 4]
```diff
@@ -1,37 +1,27 @@
...
 RUN pip install matplotlib
 
 COPY . /opt/faiss
 
 WORKDIR /opt/faiss
 
-ENV BLASLDFLAGS /usr/lib/libopenblas.so.0
+ENV BLASLDFLAGS=/usr/lib/libopenblas.so.0
 
 RUN mv example_makefiles/makefile.inc.Linux ./makefile.inc
```
Comment (beauby) [inline Dockerfile]: This is not needed anymore, since `./configure` will generate `makefile.inc`.

**Comment by mdouze** [comment]
@beauby, what should we do with this PR?

**Comment by beauby** [comment]
Thanks for your work Pierre. Closing this as a new Dockerfile is available in #769.

**Pull request closed without merging**